api/config/v1alpha1/tracing_types.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type ProxyTracing struct {
+	// RandomSamplingPercentage controls the rate at which traffic will be
+	// selected for tracing if no prior sampling decision has been made.
+	// Defaults to 0, valid values [0-100]. 100 indicates 100% sampling.
```

Comment:
should this default to 100 instead ?

Reply:
Actually, tracing may cause performance issue, particularly on edge proxy with high load, that's why I prefer to disabled it by default.
 @envoyproxy/gateway-maintainers WDYT?

Reply:
since tracing is disabled by default, I think defaulting sampling rate to 100 is okay when tracing is enabled, imo

Reply:
imo 100 by default is the least confusing. Just need to make sure we call this out in the docs in a way that is very visible so that users can turn this down if they want to.

Reply:
yes, I will update the default value and add a note in the doc.